addons/mail/static/src/js/composers/mention_manager.js

```diff
@@ -385,7 +385,7 @@ var MentionManager = Widget.extend({
             // no substitution string given, so use the mention name instead
             // replace white spaces with non-breaking spaces to facilitate
             // mentions detection in text
-            selectedSuggestion.name = selectedSuggestion.name.replace(/ /g, NON_BREAKING_SPACE);
+            selectedSuggestion.name = _.unescape(selectedSuggestion.name.replace(/ /g, NON_BREAKING_SPACE));
```

Comment:
unescaping only for UI rendering (avoid having this : "Project &amp; others" but having this instead : "Project & others")

Reply:
put following comment before this LOC:
```js
// mention manager stores escaped suggestion names
```